tools/python/boutiques/validator.py

```diff
@@ -26,10 +26,17 @@ def validate_descriptor(json_file, **kwargs):
         schema = simplejson.load(fhandle)
 
     # Load descriptor
-    with open(json_file) as fhandle:
+    json_object = kwargs.get('json_object')
+    if not json_object:
+        with open(json_file) as fhandle:
+            try:
+                descriptor = simplejson.load(fhandle)
+            except simplejson.errors.JSONDecodeError as e:
+                raise DescriptorValidationError(str(e))
+    else:
         try:
-            descriptor = simplejson.load(fhandle)
-        except simplejson.errors.JSONDecodeError as e:
+            descriptor = json.loads(json_file)
+        except ValueError as e:
```

Comment:
Create a function that will return the JSON object or the JSON (on demand) from a string representing a JSON file or object. Use this function as much as possible.